From 286122d2e5e5b662652a3a339ec1141d6b7b95cd Mon Sep 17 00:00:00 2001
From: Davor Joja <davorjoja@logicbricks.com>
Date: Thu, 3 Apr 2014 18:24:50 +0200
Subject: [PATCH 133/456] gpu: drm: xylon: Updated top driver get bpp
 functionality

This patch comes from:
  https://github.com/Xilinx/linux-xlnx.git

Changed bpp variable type in Xylon DRM driver.
Updated getting of bpp value with new functionality.

Signed-off-by: Davor Joja <davorjoja@logicbricks.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
(cherry picked from commit d6cf8452906c11f4b6b09ab33ce722bead3b7c37)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/gpu/drm/xylon/xylon_drv.c |   10 ++++++++--
 1 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/xylon/xylon_drv.c b/drivers/gpu/drm/xylon/xylon_drv.c
index 951e6a3..818ff29 100644
--- a/drivers/gpu/drm/xylon/xylon_drv.c
+++ b/drivers/gpu/drm/xylon/xylon_drv.c
@@ -47,7 +47,8 @@ static int xylon_drm_load(struct drm_device *dev, unsigned long flags)
 {
 	struct platform_device *pdev = dev->platformdev;
 	struct xylon_drm_device *xdev;
-	int ret, bpp;
+	unsigned int bpp;
+	int ret;
 
 	xdev = devm_kzalloc(dev->dev, sizeof(*xdev), GFP_KERNEL);
 	if (!xdev)
@@ -96,7 +97,12 @@ static int xylon_drm_load(struct drm_device *dev, unsigned long flags)
 		goto err_irq;
 	}
 
-	bpp = xylon_drm_crtc_get_bits_per_pixel(xdev->crtc);
+	ret = xylon_drm_crtc_get_param(xdev->crtc, &bpp,
+				       XYLON_DRM_CRTC_BUFF_BPP);
+	if (ret) {
+		DRM_ERROR("failed get bpp\n");
+		goto err_fbdev;
+	}
 	xdev->fbdev = xylon_drm_fbdev_init(dev, bpp, 1, 1);
 	if (IS_ERR(xdev->fbdev)) {
 		DRM_ERROR("failed initialize fbdev\n");
-- 
1.7.5.4

